Source/SwiftLintFramework/Rules/ValidIBInspectableRule.swift

```diff
@@ -0,0 +1,130 @@
+//
+//  ValidIBInspectableRule.swift
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 10/20/2016.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct ValidIBInspectableRule: ASTRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.Warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "valid_ibinspectable",
+        name: "Valid IBInspectable",
+        description: "@IBInspectable should be applied to variables only, have its type explicit " +
+        "and be of a supported type",
+        nonTriggeringExamples: [
+            "class Foo {\n  @IBInspectable private var x: Int\n}\n",
+            "class Foo {\n  @IBInspectable private var x: String?\n}\n",
+            "class Foo {\n  @IBInspectable private var x: String!\n}\n",
+            "class Foo {\n  @IBInspectable private var x: ImplicitlyUnwrappedOptional<String>\n}\n",
+            "class Foo {\n  @IBInspectable private var x: Optional<String>\n}\n",
+            "class Foo {\n  @IBInspectable private var count: Int = 0\n}\n",
+            "class Foo {\n  private var notInspectable = 0\n}\n",
+            "class Foo {\n  private let notInspectable: Int\n}\n",
+        ],
+        triggeringExamples: [
+            "class Foo {\n  @IBInspectable private let count: Int\n}\n",
+            "class Foo {\n  @IBInspectable private var insets: UIEdgeInsets\n}\n",
+            "class Foo {\n  @IBInspectable private var count = 0\n}\n",
+            "class Foo {\n  @IBInspectable private var count: Int?\n}\n",
+            "class Foo {\n  @IBInspectable private var count: Int!\n}\n",
+            "class Foo {\n  @IBInspectable private var x: ImplicitlyUnwrappedOptional<Int>\n}\n",
+            "class Foo {\n  @IBInspectable private var count: Optional<Int>\n}\n",
+        ]
+    )
+
+    public func validateFile(file: File,
+                             kind: SwiftDeclarationKind,
+                             dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+
+        guard kind == .VarInstance else {
+            return []
+        }
+
+        // Check if IBInspectable
+        let attributes = (dictionary["key.attributes"] as? [SourceKitRepresentable])?
+            .flatMap({ ($0 as? [String: SourceKitRepresentable]) as? [String: String] })
+            .flatMap({ $0["key.attribute"] }) ?? []
+        let isIBInspectable = attributes.contains("source.decl.attribute.ibinspectable")
+        guard isIBInspectable else {
+            return []
+        }
+
+        // if key.setter_accessibility is nil, it's a `let` declaration
```

Comment:
I've assumed this from experimentation - is there a better way to check if it's a `let`?